src/chat/MarkAsReadButton.js

```diff
@@ -0,0 +1,74 @@
+/* @flow strict-local */
+
+import React, { useCallback } from 'react';
+
+import type { Narrow } from '../types';
+import { createStyleSheet } from '../styles';
+import { useSelector } from '../react-redux';
+import { ZulipButton } from '../common';
+import * as api from '../api';
+import { getAuth, getStreams } from '../selectors';
+import {
+  isHomeNarrow,
+  isStreamNarrow,
+  isTopicNarrow,
+  streamNameOfNarrow,
+  topicOfNarrow,
+} from '../utils/narrow';
+
+const styles = createStyleSheet({
+  button: {
+    borderRadius: 16,
+    height: 32,
+    paddingLeft: 12,
+    paddingRight: 12,
+  },
+});
+
+type Props = $ReadOnly<{|
+  narrow: Narrow,
+|}>;
+
+export default function MarkAsReadButton(props: Props) {
```

Comment:
MarkAsReadButton [nfc]: Rename, from MarkUnreadButton.
    
    The button marks things as read, not unread.

:upside_down_face: :laughing: :+1: